java/src/main/java/ai/rapids/cudf/DecimalUtils.java

```diff
@@ -82,13 +82,15 @@ public static ColumnVector lessThan(ColumnView lhs, BigDecimal rhs) {
     int leftScale = lhs.getType().getScale();
     int leftPrecision = lhs.getType().getDecimalMaxPrecision();
 
-    // First we have to round the scalar (rhs) to the same scale as lhs.  Because this is a
-    // less than and it is rhs that we are rounding, we will round away from 0 (UP)
-    // to make sure we always return the correct value.
+    // First we have to round the scalar (rhs) to the same scale as lhs.
+    // Because this is a less-than, and it is rhs that we are rounding, we will round away from 0 (UP) in case rhs is
+    // positive and toward 0 (DOWN) if rhs is negative to make sure we always return the correct value.
     // For example:
-    //      100.1 < 100.19
-    // If we rounded down the rhs 100.19 would become 100.1, and now 100.1 is not < 100.1
-    BigDecimal roundedRhs = rhs.setScale(-leftScale, BigDecimal.ROUND_UP);
+    //      ex:1 100.1 < 100.19
+    //      ex:2 -100.2 < -100.19
+    // In ex:1 If we rounded down the rhs 100.19 would become 100.1, and now 100.1 is not < 100.1
+    // In ex:2 If we rounded up the rhs -100.19 would become -100.2, and now -100.2 is not < -100.2
```

Comment:
So what are the correct rounding for these examples? I'm confused. Please be explicit.